src/bp/ui-studio/src/web/views/FlowBuilder/diagram/nodes/Block/index.tsx

```diff
@@ -0,0 +1,320 @@
+  say_something: 'studio.flow.node.chatbotSays',
+  success: 'studio.flow.node.workflowSucceeds',
+  trigger: 'studio.flow.node.triggeredBy'
+}
+
+const BlockWidget: FC<Props> = ({
+  node,
+  editNodeItem,
+  onDeleteSelectedElements,
+  onCopySelectedElement,
+  selectedNodeItem,
+  getConditions,
+  switchFlowNode,
+  updateFlowNode,
+  getLanguage,
+  getExpandedNodes,
+  setExpanded,
+  getDebugInfo,
+  editTriggers,
+  disconnectNode
+}) => {
+  const { nodeType } = node
+  const { currentLang, defaultLang } = getLanguage()
+
+  const handleContextMenu = e => {
+    e.stopPropagation()
+    e.preventDefault()
+
+    if (defaultLang && defaultLang !== currentLang) {
+      toast.info('studio.flow.cannotAddContent')
+      return
+    }
+
+    switchFlowNode(node.id)
+    contextMenu(
+      e,
+      <Menu>
+        <MenuItem
+          icon="trash"
+          text={
+            <div className={sharedStyle.contextMenuLabel}>
+              {lang.tr('delete')}
+              <ShortcutLabel light keys={['backspace']} />
+            </div>
+          }
+          intent={Intent.DANGER}
+          onClick={onDeleteSelectedElements}
+        />
+        <MenuItem
+          icon="duplicate"
+          text={<div className={sharedStyle.contextMenuLabel}>{lang.tr('copy')}</div>}
+          onClick={() => onCopySelectedElement(node.id)}
+        />
+        <MenuDivider />
+        {nodeType === 'trigger' && <MenuItem icon="edit" text={lang.tr('edit')} onClick={() => editTriggers(node)} />}
+        <MenuItem icon="minimize" text={lang.tr('studio.flow.disconnectNode')} onClick={() => disconnectNode(node)} />
+        {nodeType === 'router' ? (
+          <React.Fragment>
+            <MenuDivider />
+            <MenuItem text={lang.tr('studio.flow.chips')}>
+              <MenuItem
+                text={lang.tr('studio.flow.transition')}
+                onClick={async () => {
+                  await switchFlowNode(node.id)
+                  updateFlowNode({ next: [...node.next, defaultTransition] })
+                }}
+                icon="flow-end"
+              />
+            </MenuItem>
+          </React.Fragment>
+        ) : null}
+      </Menu>
+    )
+  }
+
+  const inputPortInHeader = !['trigger'].includes(nodeType)
+  const outPortInHeader = !['failure', 'router', 'success'].includes(nodeType)
+  const canCollapse = !['failure', 'router', 'success', 'listen'].includes(nodeType)
+  const hasContextMenu = !['failure', 'success'].includes(nodeType)
+
+  const debugInfo = getDebugInfo(node.name)
+
+  const renderContents = () => {
+    switch (nodeType) {
+      case 'action':
+        return <ActionContents node={node} editNodeItem={editNodeItem} />
+      case 'execute':
+        return <ExecuteContents node={node} updateFlowNode={updateFlowNode} switchFlowNode={switchFlowNode} />
+      case 'router':
+        return <RouterContents node={node} editNodeItem={editNodeItem} selectedNodeItem={selectedNodeItem} />
```

Comment:
Is this node necessary ? since we continue to support transitions in the standard node

Reply:
It is to be backward compatible, since anyone using NDU right now has access to the router node. Same thing for listen (which we got rid in Albert)